lib/Doctrine/DBAL/Platforms/MySqlPlatform.php

```diff
@@ -1150,4 +1151,35 @@ public function getDefaultTransactionIsolationLevel()
     {
         return TransactionIsolationLevel::REPEATABLE_READ;
     }
+
+    public function supportsColumnLengthIndexes()
+    {
+        return true;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getIndexFieldDeclarationListSQL($columnsOrIndex) : string
+    {
+        if (! ($columnsOrIndex instanceof Index)) {
+            return parent::getIndexFieldDeclarationListSQL($columnsOrIndex);
+        }
+
+        $subParts = $columnsOrIndex->hasOption('lengths') ? $columnsOrIndex->getOption('lengths') : [];
+
+        $ret = [];
+
+        foreach ($columnsOrIndex->getQuotedColumns($this) as $quotedColumn) {
+            $length = array_shift($subParts);
+
+            if ($length !== null) {
```

Comment:
hum, although I do agree that it would be nice to have a base implementation in base class, current way of declaring SQL indexes with partial columns is MySQL specific and does not belong to base class